bokehjs/src/coffee/core/selection_manager.ts

```diff
@@ -20,39 +20,73 @@ export class SelectionManager extends HasProps {
     })
   }
 
-  source: DataSource
-
-  selector: Selector
-  inspectors: {[key: string]: HasProps}
+  source: ColumnarDataSource
+  inspectors: {[key: string]: Selection}
 
   initialize(): void {
     super.initialize()
-    this.selector = new Selector()
     this.inspectors = {}
   }
 
   select(renderer_views: RendererView[], geometry: Geometry, final: boolean, append: boolean = false): boolean {
-    let did_hit = false
+    // divide renderers into glyph_renderers or graph_renderers
+    const glyph_renderer_views: RendererView[] = []
+    const graph_renderer_views: RendererView[] = []
     for (const r of renderer_views) {
-      did_hit = did_hit || r.hit_test(geometry, final, append)
+      if (r.model.type == 'GlyphRenderer'){
+        glyph_renderer_views.push(r)
+      } else {
+        if (r.model instanceof GraphRenderer){
+          graph_renderer_views.push(r)
+        }
+      }
+    }
+
+    let did_hit = false
+
+    // graph renderer case
+    for (const r of graph_renderer_views) {
+      const hit_test_result = r.model.selection_policy.hit_test(geometry, r)
+      did_hit = did_hit || r.model.selection_policy.do_selection(hit_test_result, r.model, final, append)
     }
+    // glyph renderers
+    if (glyph_renderer_views.length > 0) {
+      const hit_test_result = this.source.selection_policy.hit_test(geometry, renderer_views)
+      did_hit = did_hit || this.source.selection_policy.do_selection(hit_test_result, this.source, final, append)
+    }
+
     return did_hit
   }
 
   inspect(renderer_view: RendererView, geometry: Geometry): boolean {
     let did_hit = false
-    did_hit = did_hit || renderer_view.hit_test(geometry, false, false, "inspect")
+
+    if (renderer_view.model.type == 'GlyphRenderer') {
+      const hit_test_result = renderer_view.hit_test(geometry)
+      did_hit =  !hit_test_result.is_empty()
+      const inspection = this.get_or_create_inspector(renderer_view.model)
+      inspection.update(hit_test_result, true, false)
+      this.source.setv({inspected: inspection}, {silent: true})
+      this.source.inspect.emit([renderer_view, {geometry: geometry}])
+    } else {
+      if (renderer_view.model instanceof GraphRenderer) {
+        const hit_test_result = renderer_view.model.inspection_policy.hit_test(geometry, renderer_view)
+        did_hit = did_hit || renderer_view.model.inspection_policy.do_inspection(hit_test_result, geometry, renderer_view, false, false)
+      }
+    }
+
     return did_hit
   }
 
-  clear(): void {
-    this.selector.clear()
-    this.source.selected = hittest.create_hit_test_result()
+  clear(rview: RendererView | undefined): void {
+    this.source.selected.clear()
+    if(rview)
```

Comment:
`if (view != null)` is the idiomatic approach. Also use single space between a keyword and opening parenthesis.